lib/chef/handler/datadog.rb

```diff
@@ -141,20 +141,31 @@ def prepare_the_pack
         dogs
       end
 
+      def url_or_site(value)
+        return value unless value.nil?
+
+        url = 'https://app.datadoghq.com'
+        url = 'https://app.' + @config[:site] unless @config[:site].nil?
+        url = @config[:url] unless @config[:url].nil?
+        url
+      end
+
       # return all endpoints as a list of triplets [url, api_key, application_key]
       def endpoints
         validate_keys(@config[:api_key], @config[:application_key], true)
 
-        endpoints = [[@config[:url], @config[:api_key], @config[:application_key]]]
+        # the first endpoint is always the url/site + apikey + appkey one
+        endpoints = [[url_or_site(nil), @config[:api_key], @config[:application_key]]]
 
+        # then add extra endpoints
         extra_endpoints = @config[:extra_endpoints] || []
-
         extra_endpoints.each do |endpoint|
-          url = endpoint[:url] || @config[:url]
+          url = url_or_site(endpoint[:url])
```

Comment:
I find the signature of the new `url_or_site` function a bit weird: could we simplify it to something like:

* `config_url()`, which returns the URL set in `@config` depending on the value of `@config[:site]` and ` @config[:url]`
* then, here, we can keep a logic like `url = endpoint[:url] || config_url()`

What do you think?

Reply:
You're right it looks better 👍